### PR TITLE
Sortingarticlesbydate

### PR DESCRIPTION
**Pull request by TWallwin**
added tag array property,
added sort_by, tag, order queries

### CODE REVIEW

File: __test__/app.test.js [hunk 1]
```diff
@@ -3,12 +3,15 @@ const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
 const request = require("supertest");
 const testData = require("../db/test-data/index");
-
+const { compareDates } = require("../db/helpers/utils");
+const { expect } = require("@jest/globals");
```
[the remark this answers is not in the document]
Comment (TWallwin) [reply __test__/app.test.js:7]: ah oh no

File: models/video-models.js [hunk 2]
```diff
@@ -1,34 +1,65 @@
 const db = require("../db/connection");
 
-exports.fetchVideos = () => {
-  return db.query("SELECT * FROM videos").then((result) => {
+exports.fetchVideos = (sort_by = "created_at", tags, order) => {
```
Comment (davidptclark) [inline models/video-models.js:3]: order doesn't have a default but I remember seeing that you set order=DESC is order isn't given as a query: just checking I've got that right.

File: __test__/app.test.js [hunk 3]
```diff
@@ -23,11 +26,116 @@ describe("/api/videos ", () => {
...
+
+  describe("queries:", () => {
+    describe("order", () => {
+      test("status 200 - accepts order query by asc, desc (default)", () => {
+        return request(app)
+          .get("/api/videos?order=asc")
+          .expect(200)
+          .then(({ body: { videos } }) => {
+            expect(videos).toBeSortedBy("created_at", {
+              compare: compareDates,
```
[the remark this answers is not in the document]
Comment (TWallwin) [reply __test__/app.test.js:77]: yep

File: models/video-models.js (hunk 2 above)
Comment (TWallwin) [reply models/video-models.js:3]: yeah if order is undefined in the controller function it's set to DESC before being passed to the model

File: __test__/app.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (TWallwin) [reply __test__/app.test.js]: tidied it not

**Review by davidptclark — APPROVED** [review]
Fantastic work, Tom!

**Pull request merged**